packages/markers/src/common/marker.ts

```diff
@@ -37,3 +39,15 @@ export interface Marker<T> {
      */
     data: T;
 }
+export namespace Marker {
+    export function is(value: unknown): value is Marker<object>;
+    export function is<T>(value: unknown, subTypeCheck: (value: unknown) => value is T): value is Marker<T>;
```

Comment:
I'm more worried about the function parameter, which is unusual.

Reply:
What worries you?

Reply:
Why add an overload that is not used? It makes the code more complex and currently adds nothing.

Reply:
Why: it is the proper way to test this generic type. The `Marker<object>` is just a more common occurrence, and the signature line 43 is here for convenience. If anything, that same signature would be the one that's extra.

Otherwise we have a type `Marker<T>` that could be `Marker<Potato>`, wouldn't it be weird if the only `Marker.is` function only checks for `Marker<object>`?

It would make more sense for `Marker.is` to only check for `Marker<object>` if `T` was constrained to `object` too, but that's currently not the case.

Tell me if you want it removed, otherwise I'll merge this PR.

Reply:
I want to convince you we should not add stuff that is not needed. Since it's unused, it's not needed. Nobody is passing in the `subTypeCheck` parameter, as far as I can tell.

Reply:
I introduced a new change, tell me if it's ok with you.

Reply:
> I want to convince you we should not add stuff that is not needed.

I don't think I can be convinced about taking shortcuts. The implementation with the `subTypeCheck` function is the correct implementation to type guard this type. Anything else is a specialization and might be optimized subsequently without getting rid of the original proper implementation.